Create documentation for bundles docs (#3804)
kyma-project/kyma: docs/helm-broker/docs/03-01-create-bundles.md

>**NOTE:** For more information about schemas, see [this](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#schemas-object) specification.

## docs directory

In the `docs` directory we specify documentation for the bundle and for the wrapped Helm Chart. You can define a `meta.yaml` file inside the `docs` folder which holds the information on how documentation for a bundle should be uploaded.
Currently we provide the documentation for bundle using the ClusterDocsTopics because in Kyma the Helm Broker is installed as a ClusterServiceBroker. For more information about ClusterDocsTopics see the following [file](https://kyma-project.io/docs/components/headless-cms/#custom-resource-clusterdocstopic).
If the `docs/meta.yaml` is specified the Helm Broker tries to create the ClusterDocsTopic for this bundle. Below you can see the example structure of the `meta.yaml` file.
```
docs:

>     - template:                     # template describes the ClusterDocsTopic that will be created.

Reviewer: we provided an example, but where I can learn more information about which fields I can provide?

Author: I added a sentence about this below the example

        displayName: "Doc for redis bundle"
        description: "Overall documentation"
        sources:                    # list of files to upload as a Asset
          - type: markdown
            name: markdown-files    # must be unique within sources
            mode: package           # defines that file under below 
            url: abc.pl             # if not provided then Helm broker will inject the full repository URL for this bundle
            filter: /docs/bundles
```
For now, we support only one entry in the `docs` array.

> The Helm Broker can provision a broker which provides its own ServiceClasses. You can see how to provide a documentation for them in the following [document](#details-bundles-docs).

Author: This link will work on kyma-project.io, we must define it in this way because of governance job which blocks links to the not existing documents.

>**NOTE:** For more information about schemas, see [this](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#schemas-object) specification.

## docs directory

In the `docs` directory we specify documentation for the bundle and for the wrapped Helm Chart. You can define a `meta.yaml` file inside the `docs` folder which holds the information on how documentation for a bundle should be uploaded.
Currently we provide the documentation for bundle using the ClusterDocsTopics because in Kyma the Helm Broker is installed as a ClusterServiceBroker.
If the `docs/meta.yaml` is specified the Helm Broker tries to create the ClusterDocsTopic for this bundle. Below you can see the example structure of the `meta.yaml` file.
```
docs:

>     - template:                     # template describes the ClusterDocsTopic that will be created.

Reviewer 2: ```suggestion
    - template:                     # A template which describes the ClusterDocsTopic that will be created
```

>         displayName: "Doc for redis bundle"

Reviewer 2: ```suggestion
        displayName: "Docs for the Redis bundle"
```

        description: "Overall documentation"

>         sources:                    # list of files to upload as a Asset

Reviewer 2: ```suggestion
        sources:                    # A list of files that are uploaded as assets
```

          - type: markdown

>             name: markdown-files    # must be unique within sources

Reviewer 2: ```suggestion
            name: markdown-files    # A name that must be unique within sources
```

>             mode: package           # defines that file under below 

Reviewer 2: What?

Author: fixed xD

>             url: abc.pl             # if not provided then Helm broker will inject the full repository URL for this bundle

Reviewer 2: ```suggestion
            url: abc.pl             # A URL to the documentation. If not provided, the Helm broker will inject the full repository URL for this bundle.
```

            filter: /docs/bundles
```

> For more information about the fields you can provide and ClusterDocsTopics see the following [file](https://kyma-project.io/docs/components/headless-cms/#custom-resource-clusterdocstopic).

Reviewer 2: Remove this sentence and provide the link to the CLuterDocsTopics in a place when you mention it for the first time.

Author: done

> For now, we support only one entry in the `docs` array.

Reviewer 2: ```suggestion
>**NOTE:** You can provide only one entry in the `docs` array.
```

> The Helm Broker can provision a broker which provides its own ServiceClasses. You can see how to provide a documentation for them in the following [document](./03-04-bundles-docs.md).

Reviewer 2: ```suggestion
The Helm Broker can provision a broker which provides its own ServiceClasses. To learn how to upload documentation for those classes, read [this](#details-bundles-docs.md) document.
```
